### ISSUE TITLE
Go module keeps pulling non existent module

### ISSUE THREAD

**Issue by dadepo** [golang/go#38664]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.14.1 darwin/amd64

</pre>

### Does this issue reproduce with the latest release?

yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env

Oluwadadepos-MacBook-Air:go-ipfs oluwadadepoaderemi$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/oluwadadepoaderemi/Library/Caches/go-build"
GOENV="/Users/oluwadadepoaderemi/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/oluwadadepoaderemi/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/oluwadadepoaderemi/Documents/masters/ot/project/go-ipfs/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/09/k14t2v457rgbn98cgg3kjc100000gn/T/go-build330566292=/tmp/go-build -gno-record-gcc-switches -fno-common"

</pre></details>

### What did you do?

working with a project that depends on a forked version of its module 
clean module cache

i run `go mod tidy`

it gives the output:

```
Oluwadadepos-MacBook-Air:go-ipfs oluwadadepoaderemi$ go mod tidy
go: finding module for package github.com/dadepo/go-ipfs-config
go: found github.com/dadepo/go-ipfs-config in github.com/dadepo/go-ipfs-config v0.5.3
```

*Ediit:* Also when I check the dependency by running go mod graph I get:

```
Oluwadadepos-MacBook-Air:go-ipfs oluwadadepoaderemi$ go mod graph | grep "github.com/dadepo/go-ipfs-config"
github.com/ipfs/go-ipfs github.com/dadepo/go-ipfs-config@v0.5.3
github.com/dadepo/go-ipfs-config@v0.5.3 github.com/facebookgo/atomicfile@v0.0.0-20151019160806-2de1f203e7d5
github.com/dadepo/go-ipfs-config@v0.5.3 github.com/libp2p/go-libp2p-core@v0.5.1
github.com/dadepo/go-ipfs-config@v0.5.3 github.com/mitchellh/go-homedir@v1.1.0
github.com/dadepo/go-ipfs-config@v0.5.3 github.com/multiformats/go-multiaddr@v0.2.1
github.com/dadepo/go-ipfs-config@v0.5.3 github.com/multiformats/go-multibase@v0.0.2
github.com/dadepo/go-ipfs-config@v0.5.3 github.com/smola/gocompat@v0.2.0
github.com/dadepo/go-ipfs-config@v0.5.3 golang.org/x/crypto@v0.0.0-20200414173820-0848c9571904
github.com/dadepo/go-ipfs-config@v0.5.3 golang.org/x/sys@v0.0.0-20200413165638-669c56c373c4
```
and also when I run `go build` the `go.mod` file is updated with this line added:

```
	github.com/dadepo/go-ipfs-config v0.5.3 // indirect
```

Also when I list the versions of github.com/dadepo/go-ipfs-config I get the following

```
github.com/dadepo/go-ipfs-config v0.0.1 v0.0.2 v0.0.3 v0.0.4 v0.0.5 v0.0.6 v0.0.7 v0.0.8 v0.0.9 v0.0.10 v0.0.11 v0.0.12 v0.0.30 v0.0.31 v0.0.33 v0.1.0 v0.2.0 v0.2.1 v0.3.0 v0.4.0 v0.4.1 v0.4.2 v0.5.0 v0.5.1 v0.5.2 v0.5.3
```

Which includes v0.5.3 which is a version does not exist (unless I am missing something)

it should be noted I have in go.mod the following:

```
replace github.com/ipfs/go-ipfs-config v0.5.2 => github.com/dadepo/go-ipfs-config v0.0.31
```

### What did you expect to see?

it should not include v0.5.3 of github.com/dadepo/go-ipfs-config as it does not exist. This can be confirmed by checking the repo https://github.com/dadepo/go-ipfs-config no branch or tag of v0.5.3 there

### What did you see instead?

it includes a non existent dependency

**Comment by seankhliao** [comment]
version 0.5.3 exists in the proxy, added 2020-04-20 https://proxy.golang.org/github.com/dadepo/go-ipfs-config/@v/v0.5.3.info

was history rewritten?

**Comment by dadepo** [comment]
> was history rewritten?

Probably. Just picking up Go and I fumbled about for a quite a bit trying to understand how to use the module system - so I really cannot remember all the things I did, did not do, tripped over, pushed aside (you get the idea) in the process. I think at some point I created the branch/tag then deleted it. 

> version 0.5.3 exists in the proxy, added 2020-04-20 https://proxy.golang.org/github.com/dadepo/go-ipfs-config/@v/v0.5.3.info

oly cow! this really got me bewildered for a while. So what is this proxy thing? How could I have discovered it myself? And more importantly how do I clear it. I ran go `go clean -modcache` a couple of time, but guess that was only dealing with stuff on my machine.

**Comment by dadepo** [comment]
Ok, found how to switch the proxy off via GOPROXY environment variable. That should do the trick.

**Comment by dadepo** [comment]
Ok, so it turns out I was pulling a number of noob moves, and getting this wrong. 

For someone else who stumbles on this issue, perhaps the following would help:

1. https://proxy.golang.org is a thing! If you made a release, deleted it, chances are that the version is already cached in  https://proxy.golang.org. Check this if you are seeing version arising from the dead and you wondering how come. This documentation [link](https://golang.org/cmd/go/#hdr-Module_downloading_and_verification) also sheds more light on the proxy.
2. When forking a dependency, there is no need to update the module path in the forked version. In my case, I forked `github.com/ipfs/go-ipfs-config` to `github.com/dadepo/go-ipfs-config` and I also updated the module path to `github.com/dadepo/go-ipfs-config`. That was a wrong move that lead to other issues.

Closing as this is not a bug, but an end-user problem.

Thanks @seankhliao for the help.
